libvast/src/system/signal_reflector.cpp

```diff
@@ -0,0 +1,74 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2022 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "vast/system/signal_reflector.hpp"
+
+#include <caf/blocking_actor.hpp>
+
+#include <csignal>
+
+namespace vast::system {
+
+namespace {
+
+void signal_listener(caf::blocking_actor* self,
+                     const signal_reflector_actor& reflector, sigset_t sigset) {
+  int signum = 0;
+  sigwait(&sigset, &signum);
+  self->send<caf::message_priority::high>(reflector, atom::internal_v,
+                                          atom::signal_v, signum);
+}
+
+} // namespace
+
+sigset_t termsigset() {
+  sigset_t sigset;
+  sigemptyset(&sigset);
+  sigaddset(&sigset, SIGINT);
+  sigaddset(&sigset, SIGTERM);
+  return sigset;
+}
+
+signal_reflector_actor::behavior_type signal_reflector(
+  signal_reflector_actor::stateful_pointer<signal_reflector_state> self,
+  sigset_t sigset) {
+  self->set_exit_handler([=](const caf::exit_msg&) {
+    self->state.stop = true;
+    kill(getpid(), SIGTERM);
```

Comment:
I'm not sure this is correct. On this branch, if I press C-c once, `vast start` exits with exit code 143. That should be zero instead.

Reply:
This now works as expected.